### PR TITLE
Fix this_week filter [SDESK-7264]

### PR DESCRIPTION
**Pull request by devketanpro**
Fix items from previous week along with items from this week are also displayed

### CODE REVIEW

File: server/planning/search/queries/elastic.py [hunk 1]
```diff
@@ -276,7 +279,7 @@ def range_this_week(query: ElasticRangeParams):
             time_zone=query.time_zone,
             value_format=query.value_format,
             gte=start_of_this_week(query.start_of_week),
-            lt=start_of_next_week(query.start_of_week),
+            lt=end_of_this_week(query.start_of_week),
```
[the remark this answers is not in the document]
Comment (devketanpro) [reply server/planning/search/queries/elastic.py:282]: no this is not ignoring events on that last date see :-
image

[the remark this answers is not in the document]
Comment (petrjasek) [reply server/planning/search/queries/elastic.py:282]: but looking at the test the query would be like `<2024-05-19T00:00:00` which seems wrong, question is why that event is visible there but I can't see the timestamps, might be due to the timezone

[the remark this answers is not in the document]
Comment (devketanpro) [reply server/planning/search/queries/elastic.py]: yep we have timezone issue,and when we search from sidenav bar filters work perfectly but if we using events&planning filters we got wrong items

File: server/planning/search/queries/combined.py [hunk 2]
```diff
@@ -22,8 +22,12 @@ def construct_combined_view_data_query(
 
     query = elastic.ElasticQuery()
 
-    if len(search_filter["params"]):
-        search_dates(search_filter["params"], query)
+    filter_params = common.get_params_from_search_filter(search_filter)
+    if len(filter_params):
+        filter_params["time_zone"] = params.get("time_zone")
+        filter_params["start_of_week"] = params.get("start_of_week")
```
Comment (petrjasek) [inline server/planning/search/queries/combined.py:28]: I wonder how this will be populated, especially when using those search filters for scheduled exports. imo those values are not stored in search filters atm

Comment (MarkLark86) [reply server/planning/search/queries/combined.py:28]: What about add a `time_zone` input for scheduled exports form, and use app config for `start_of_week` on search?

File: server/planning/search/queries/elastic.py [hunk 3]
```diff
@@ -120,12 +120,15 @@ def __init__(
...
-    end = get_start_of_next_week(date, start_of_week) - timedelta(days=7)
-    start = end - timedelta(days=7)
-
+    start = get_start_of_next_week(date, start_of_week) - timedelta(days=7)
     return start.strftime("%Y-%m-%d") + "||/d"
 
 
+def end_of_this_week(start_of_week=0, date=None):
+    end = get_start_of_next_week(date, start_of_week) - timedelta(start_of_week)
+    return end.strftime("%Y-%m-%d") + "||/d"
```
[the remark this answers is not in the document]
Comment (devketanpro) [reply server/planning/search/queries/elastic.py]: when we send the full datetime, all filters misbehave, resulting in items with different dates.

File: server/planning/search/queries/combined.py (hunk 2 above)
Comment (petrjasek) [reply server/planning/search/queries/combined.py:28]: we have also default timezone config, I would probably start with that, not sure if we have a need to use different timezones in those exports

File: server/planning/search/queries/combined.py [hunk 4]
```diff
@@ -22,8 +23,12 @@ def construct_combined_view_data_query(
 
     query = elastic.ElasticQuery()
 
-    if len(search_filter["params"]):
-        search_dates(search_filter["params"], query)
+    filter_params = common.get_params_from_search_filter(search_filter)
+    if len(filter_params):
+        filter_params["time_zone"] = params.get("time_zone") or app.config.get("DEFAULT_TIMEZONE")
+        filter_params["start_of_week"] = params.get("start_of_week") or app.config.get("START_OF_WEEK", 0)
```
Comment (petrjasek) [inline server/planning/search/queries/combined.py:29]: this might be tricky if we start using the params for it and there will be `0`, rather do it like:
```py
params.get("start_of_week", app.config.get("START_OF_WEEK", 0))
```

File: server/planning/tests/utils_tests.py [hunk 5]
```diff
@@ -62,3 +62,12 @@ def test_end_of_next_week(self):
         end_date_str = elastic.end_of_next_week(date=start_date, start_of_week=1)
         expected_weekday = self.get_start_date_and_weekday(end_date_str)
         self.assertEqual(expected_weekday, 0)  # Monday
+
+    def test_events_within_current_week(self):
+        # Test events that start and end within the current week
+        start_date = datetime(2024, 5, 15)  # Assuming today is May 15, 2024 (Wed)
```
Comment (petrjasek) [inline server/planning/tests/utils_tests.py:68]: pls add a few more cases, especially for edge cases like monday/sunday

File: server/planning/tests/utils_tests.py [hunk 6]
```diff
@@ -71,3 +71,21 @@ def test_events_within_current_week(self):
...
+        self.assertEqual(end, "2024-05-20||/d")
+
+    def test_events_within_current_week_sunday(self):
+        # Test case for Sunday
+        start_date = datetime(2024, 5, 19)  # May 19, 2024 is a Sunday
+        start = elastic.start_of_this_week(date=start_date, start_of_week=1)
+        end = elastic.start_of_next_week(date=start_date, start_of_week=1)
+
+        self.assertEqual(start, "2024-05-20||/d")
+        self.assertEqual(end, "2024-05-27||/d")
```
Comment (petrjasek) [inline server/planning/tests/utils_tests.py:91]: I think we should get rid of that `//d`

File: server/planning/tests/utils_tests.py [hunk 7]
```diff
@@ -69,23 +69,23 @@ def test_events_within_current_week(self):
...
 
     def test_events_within_current_week_sunday(self):
         # Test case for Sunday
         start_date = datetime(2024, 5, 19)  # May 19, 2024 is a Sunday
         start = elastic.start_of_this_week(date=start_date, start_of_week=1)
         end = elastic.start_of_next_week(date=start_date, start_of_week=1)
 
-        self.assertEqual(start, "2024-05-20||/d")
-        self.assertEqual(end, "2024-05-27||/d")
+        self.assertEqual(start, "2024-05-20")
```
Comment (petrjasek) [inline server/planning/tests/utils_tests.py:90]: imo this should be 13.5.

File: server/planning/tests/utils_tests.py [hunk 8]
```diff
@@ -69,23 +69,23 @@ def test_events_within_current_week(self):
...
+        self.assertEqual(end, "2024-05-27")
```
Comment (petrjasek) [inline server/planning/tests/utils_tests.py:91]: and this one should be 20.5.

Comment (devketanpro) [reply server/planning/tests/utils_tests.py]: thanks for pointing

**Review by petrjasek — APPROVED** [review]
